src/checkqueue.h

```diff
@@ -167,15 +167,20 @@ class CCheckQueue
     //! Add a batch of checks to the queue
     void Add(std::vector<T>& vChecks)
     {
-        LOCK(m_mutex);
-        for (T& check : vChecks) {
-            queue.push_back(T());
-            check.swap(queue.back());
+        unsigned int num_of_new_checks;
+        {
+            LOCK(m_mutex);
+            for (T& check : vChecks) {
+                queue.push_back(T());
```

Comment:
nit: maybe not part of this PR, but using `queue.emplace_back()` instead `queue.push_back(T())` would save one move/copy constructor and one destructor because the object could be directly constructed in the queue

Reply:
What about `queue.push_back(std::move(check))`?

Reply:
I tried that once but then tests fail, I think for the types used here `std::swap` is implemented but a move constructor isn't. I didn't have a closer look why that fails.

I think the nicest c++ solution would be to replace the whole loop with

```cpp
queue.insert(queue.end(), std::make_move_iterator(vChecks.begin()), std::make_move_iterator(vChecks.end()));
```